From 84fc8a50f692624921d4233dec1a3c8796f5b5da Mon Sep 17 00:00:00 2001
From: Holger Hans Peter Freyther <holger@freyther.de>
Date: Mon, 26 May 2014 07:53:05 +0200
Subject: blox: Direct usage of result is deprecated and stops working

Use Tcl_GetStringResult(interp) instead of interp->result on
newer versions of Tcl. It looks like Tcl_GetStringResult has
been present in the entire 8.0 series.

2014-05-26  Holger Hans Peter Freyther  <holger@moiji-mobile.com>

	* BloxTK.c: Use Tcl_GetStringResult to access the result.
---
 packages/blox/tk/BloxTK.c  | 4 ++--
 packages/blox/tk/ChangeLog | 4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/packages/blox/tk/BloxTK.c b/packages/blox/tk/BloxTK.c
index 2ba40b81..2f06b7f0 100644
--- a/packages/blox/tk/BloxTK.c
+++ b/packages/blox/tk/BloxTK.c
@@ -173,12 +173,12 @@ tclInit (void)
 
   if (Tcl_Init (interp) == TCL_ERROR)
     {
-      fprintf (stderr, "Tcl_Init failed: %s\n", interp->result);
+      fprintf (stderr, "Tcl_Init failed: %s\n",  Tcl_GetStringResult(interp));
       exit (1);
     }
   if (Tk_Init (interp) == TCL_ERROR)
     {
-      fprintf (stderr, "Tk_Init failed: %s\n", interp->result);
+      fprintf (stderr, "Tk_Init failed: %s\n", Tcl_GetStringResult(interp));
       exit (1);
     }
   Tcl_CreateCommand (interp, "callback", doCallback, NULL, NULL);
diff --git a/packages/blox/tk/ChangeLog b/packages/blox/tk/ChangeLog
index aefd2dcb..159b1776 100644
--- a/packages/blox/tk/ChangeLog
+++ b/packages/blox/tk/ChangeLog
@@ -1,3 +1,7 @@
+2014-05-26  Holger Hans Peter Freyther  <holger@moiji-mobile.com>
+
+	* BloxTK.c: Use Tcl_GetStringResult to access the result.
+
 2010-12-04  Paolo Bonzini  <bonzini@gnu.org>
 
 	* package.xml: Remove now superfluous <file> tags.
-- 
2.11.0

